src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -292,6 +293,7 @@ def on_save_checkpoint(self) -> Dict:
     def on_load_checkpoint(self, state_dict: Dict) -> None:
         # cache the dataloader state dict until the dataloader objects are available
         self._dataloader_state_dict = state_dict.get("dataloader_state_dict")
+        self._batches_that_stepped = state_dict.get("_batches_that_stepped")
```

Comment:
How is mypy passing this test? _batches_that_stepped here is an int, but get() can return None.

```suggestion
        self._batches_that_stepped = state_dict.get("_batches_that_stepped", 0)
```